File: routes/proofIdRoutes.js
const express = require('express');
const router = express.Router();
const proofIdController = require('../controllers/proofIdController');

// Create a new proof ID
router.post('/', proofIdController.createProofId);

// Get all proof IDs
router.get('/', proofIdController.getAllProofIds);

// Get a proof ID by ID
router.get('/:id', proofIdController.getProofIdById);

// Update a proof ID by ID
router.patch('/:id', proofIdController.updateProofId);

// Delete a proof ID by ID
router.delete('/:id', proofIdController.deleteProofId);

module.exports = router;

File: routes/proof_id_routes.py
from fastapi import APIRouter
from controllers.proof_id_controller import ProofIdController

router = APIRouter(prefix="/proof-ids", tags=["Proof IDs"])

@router.post("/", response_model=dict)
async def create_proof_id(data: dict):
    return await ProofIdController.create_proof_id(data)

@router.get("/", response_model=list)
async def get_all_proof_ids():
    return await ProofIdController.get_all_proof_ids()

@router.get("/{id}", response_model=dict)
async def get_proof_id_by_id(id: str):
    return await ProofIdController.get_proof_id_by_id(id)

@router.patch("/{id}", response_model=dict)
async def update_proof_id(id: str, data: dict):
    return await ProofIdController.update_proof_id(id, data)

@router.delete("/{id}", response_model=dict)
async def delete_proof_id(id: str):
    return await ProofIdController.delete_proof_id(id)